File: Ram/Backend_Ram/Backend_Ram/Services/AdminService.cs
public class AdminService : IAdminService
{
    private readonly AppDbContext _context;

    public AdminService(AppDbContext context)
    {
        _context = context;
    }

    public List<Appointment> GetAllAppointments()
    {
        return _context.Appointments.ToList();
    }

    public string Approve(int appointmentId)
    {
        var appt = _context.Appointments.Find(appointmentId);
        if (appt == null) return "Not Found";

        appt.Status = "Approved";
        _context.SaveChanges();
        return "Approved";
    }

    public string Reject(int appointmentId)
    {
        var appt = _context.Appointments.Find(appointmentId);
        if (appt == null) return "Not Found";

        appt.Status = "Rejected";
        _context.SaveChanges();
        return "Rejected";
    }

    public string Complete(int appointmentId)
    {
        var appt = _context.Appointments.Find(appointmentId);
        if (appt == null) return "Not Found";

        appt.Status = "Completed";
        _context.SaveChanges();
        return "Completed";
    }
}

File: Ram/Frontend/src/app/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface Appointment {
  appointmentId: number;
  userId: number;
  doctorId: number;
  dateTime: string;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  private apiUrl = environment.apiUrl;

  constructor(private http: HttpClient) {}

  private getHeaders(): HttpHeaders {
    const token = localStorage.getItem('token');
    return new HttpHeaders({
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json'
    });
  }

  getAllAppointments(): Observable<Appointment[]> {
    return this.http.get<Appointment[]>(
      `${this.apiUrl}/admin/appointments`,
      { headers: this.getHeaders() }
    );
  }

  approveAppointment(id: number): Observable<string> {
    return this.http.put(
      `${this.apiUrl}/admin/appointment/approve?id=${id}`,
      {},
      { headers: this.getHeaders(), responseType: 'text' }
    );
  }

  rejectAppointment(id: number): Observable<string> {
    return this.http.put(
      `${this.apiUrl}/admin/appointment/reject?id=${id}`,
      {},
      { headers: this.getHeaders(), responseType: 'text' }
    );
  }

  completeAppointment(id: number): Observable<string> {
    return this.http.put(
      `${this.apiUrl}/admin/appointment/complete?id=${id}`,
      {},
      { headers: this.getHeaders(), responseType: 'text' }
    );
  }
}